File: backend/app/schemas/order.py
from datetime import date
from decimal import Decimal
from enum import Enum

from pydantic import BaseModel


class OrderStatus(str, Enum):
    PENDING = "PENDING"
    APPROVED = "APPROVED"
    PROCESSING = "PROCESSING"
    SHIPPED = "SHIPPED"
    DELIVERED = "DELIVERED"
    CANCELLED = "CANCELLED"


class OrderCreate(BaseModel):
    order_number: str
    company_id: int
    employee_id: int
    campaign_id: int
    order_date: date
    total_amount: Decimal
    status: OrderStatus = OrderStatus.PENDING


class OrderUpdate(BaseModel):
    order_number: str | None = None
    company_id: int | None = None
    employee_id: int | None = None
    campaign_id: int | None = None
    order_date: date | None = None
    total_amount: Decimal | None = None
    status: OrderStatus | None = None


class OrderResponse(BaseModel):
    id: int
    order_number: str
    company_id: int
    employee_id: int
    campaign_id: int
    order_date: date
    total_amount: Decimal
    status: OrderStatus

    class Config:
        from_attributes = True

File: frontend/src/api/order.ts
import api from "../services/api";

export interface Order {
  id?: number;
  order_number: string;
  company_id: number;
  employee_id: number;
  campaign_id: number;
  order_date: string;
  total_amount: number;
  status: "PENDING" | "APPROVED" | "PROCESSING" | "SHIPPED" | "DELIVERED" | "CANCELLED";
}

export const getOrders = async () => {
  const res = await api.get("/orders");
  return res.data;
};

export const getOrder = async (id: number) => {
  const res = await api.get(`/orders/${id}`);
  return res.data;
};

export const createOrder = async (data: Order) => {
  const res = await api.post("/orders", data);
  return res.data;
};

export const updateOrder = async (id: number, data: Partial<Order>) => {
  const res = await api.put(`/orders/${id}`, data);
  return res.data;
};

export const deleteOrder = async (id: number) => {
  const res = await api.delete(`/orders/${id}`);
  return res.data;
};

export const downloadInvoice = async (orderId: number) => {
  const res = await api.get(`/orders/${orderId}/invoice`, {
    responseType: "blob",
  });
  return res.data;
};
